src/pages/user/UserInstitution.tsx

```diff
@@ -1,167 +1,97 @@
 
-  const toggleOpen = () => {
-    setOpen(!open);
+  const toggleUnitForm = () => {
+    setOpenUnitForm(!openUnitForm);
   };
 
-  // const addOrgunitIdToAuthorityAndDispatchNotification = async (id: string, scn: string) => {
-  //   const updatedAuthority = await addQualifierIdForAuthority(scn, AuthorityQualifiers.ORGUNIT_ID, id);
-  //   if (updatedAuthority.error) {
-  //     dispatch(setNotification(updatedAuthority.error, NotificationVariant.Error));
-  //   } else if (updatedAuthority) {
-  //     dispatch(setAuthorityData(updatedAuthority));
-  //   }
-  // };
+  const handleSubmit = async (value: FormikInstitutionUnit) => {
+    if (!value.unit) {
+      return;
+    }
 
-  // const handleAddInstitution = async ({ id, subunits }: FormikInstitutionUnit) => {
-  //   try {
-  //     if (authority) {
-  //       if (subunits.length === 0) {
-  //         await addOrgunitIdToAuthorityAndDispatchNotification(id, authority.systemControlNumber);
-  //       } else {
-  //         const lastSubunit = subunits.pop();
-  //         await addOrgunitIdToAuthorityAndDispatchNotification(lastSubunit!.id, authority.systemControlNumber);
-  //       }
-  //     }
-  //   } catch (error) {
-  //     dispatch(setNotification(t('feedback:error.update_authority'), NotificationVariant.Error));
-  //   }
-  // };
+    const mostSpecificUnit = getMostSpecificUnit(value.unit);
+    const newUnitId = mostSpecificUnit.id.split('/').pop();
 
-  const handleSubmit = async (values: FormikInstitutionUnit, { resetForm }: any) => {
-    // if (editMode && values.editId && authority) {
-    //   const organizationUnitId = values.subunits.length > 0 ? values.subunits.pop()!.id : values.id;
-    //   const updatedAuthority = await updateQualifierIdForAuthority(
-    //     authority.systemControlNumber,
-    //     AuthorityQualifiers.ORGUNIT_ID,
-    //     values.editId,
-    //     organizationUnitId
-    //   );
-    //   if (updatedAuthority.error) {
-    //     dispatch(setNotification(updatedAuthority.error, NotificationVariant.Error));
-    //   } else if (updatedAuthority) {
-    //     dispatch(setAuthorityData(updatedAuthority));
-    //     dispatch(setNotification(t('feedback:success.update_identifier')));
-    //   }
-    // } else {
-    //   await handleAddInstitution({
-    //     name: values.name,
-    //     id: values.id,
-    //     subunits: values.subunits,
-    //     unit: values.unit,
-    //   });
-    //   resetForm(emptyFormikUnit);
-    // }
-    // setOpen(false);
-    // setEditMode(false);
-  };
+    if (!newUnitId) {
+      return;
+    } else if (authority?.orgunitids.includes(newUnitId)) {
+      dispatch(setNotification(t('feedback:info.affiliation_already_exists'), NotificationVariant.Info));
+      return;
+    }
 
-  const handleEdit = () => {
-    setOpen(true);
-    setEditMode(true);
+    if (authority) {
+      const updatedAuthority = await addQualifierIdForAuthority(
+        authority?.systemControlNumber,
```

Comment:
trenger ikke spørsmålstegn når du allerede har sjekket på authority
```suggestion
        authority.systemControlNumber,
```